api/v1alpha1/capiprovider_types.go

```diff
@@ -0,0 +1,136 @@
+//
+//nolint:lll
+type CAPIProviderSpec struct {
+	// Name is the name of the provider to enable
+	// +required
+	// +kubebuilder:validation:Enum=aws;azure;gcp;docker;rke2
+	Name ProviderName `json:"name"`
+
+	// Type is the type of the provider to enable
+	// +required
+	// +kubebuilder:validation:Enum=infrastructure;core;controlPlane;bootstrap;addon
+	Type ProviderType `json:"type"`
+
+	// Credentials is the structure holding the credentials to use for the provider.
+	// +optional
+	Credentials *ProviderCredentials `json:"credentials,omitempty"`
+
+	// Features is a collection of features to enable.
+	// +optional
+	Features *Features `json:"features,omitempty"`
+
+	// Variables is a map of environment variables to add to the content of the ConfigSecret
+	// +optional
+	Variables map[string]string `json:"variables"`
+
+	// ProviderSpec is the spec
+	ProviderSpec *operatorv1.ProviderSpec `json:",inline"`
+}
+
+// Features defines a collection of features for the CAPI Provider to apply.
+type Features struct {
+	// MachinePool if set to true will enable the machine pool. feature
+	MachinePool bool `json:"machinePool,omitempty"`
+
+	// ClusterResourceSet if set to true will enable the cluster resource set feature
+	ClusterResourceSet bool `json:"clusterResourceSet,omitempty"`
+
+	// ClusterTopology if set to true will enable the clusterclass feature.
+	ClusterTopology bool `json:"clusterTopology,omitempty"`
+}
+
+// ProviderCredentials defines the external credentials information for the provider.
+// +kubebuilder:validation:MaxProperties=1
+// +kubebuilder:validation:MinProperties=1
+// +structType=atomic
+//
+//nolint:godot
+type ProviderCredentials struct {
+	// RancherCloudCredential is the Rancher Cloud Credential name
+	RancherCloudCredential string `json:"rancherCloudCredential,omitempty"`
+
+	// +optional
+	// TODO: decide how to handle workload identity
+	// WorkloadIdentityRef *WorkloadIdentityRef `json:"workloadIdentityRef,omitempty"`
+}
+
+// WorkloadIdentityRef is a reference to an identity to be used when reconciling the cluster.
+type WorkloadIdentityRef struct {
+	// Name of the identity
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name"`
+
+	// Kind of the identity
+	Kind string `json:"kind"`
+}
+
+// CAPIProviderStatus defines the observed state of CAPIProvider.
+type CAPIProviderStatus struct {
+	// Indicates the provider status
+	// +kubebuilder:default=Pending
+	State ProviderState `json:"state,omitempty"`
+
+	// Variables is a map of environment variables added to the content of the ConfigSecret
+	// +kubebuilder:default={CLUSTER_TOPOLOGY:"true",EXP_CLUSTER_RESOURCE_SET:"true",EXP_MACHINE_POOL: "true"}
+	Variables map[string]string `json:"variables,omitempty"`
+
+	ProviderStatus *operatorv1.ProviderStatus `json:",inline"`
+}
```

Comment:
Do you think we also need conditions?

Reply:
CAPI Operator status ships with conditions, we can extend proxied list of those with additional ones.